components/Image/image-responsive.tsx

```diff
@@ -0,0 +1,288 @@
+        break
+    }
+  }
+
+  useEffect(() => {
+    setImageSource()
+    if (imgRef.current?.complete) {
+      //always need to check if image has actually loaded (given NextJS behavior)
+      handleImageLoad()
+      setIsLoaded(true)
+    }
+  }, [])
+
+  useEffect(() => {}, [scWidth])
+
+  /* POINT AND DISTANCE CALCULATION FUNCTIONS*/
+
+  //function called when we detect more than one touch. Assuming pinch only between first two touched detected
+  function distance(e: any) {
+    let zw = e.touches[0].pageX - e.touches[1].pageX
+    let zh = e.touches[0].pageY - e.touches[1].pageY
+    if (zw * zw + zh * zh != 0) {
+      return Math.sqrt(zw * zw + zh * zh)
+    } else return 0
+  }
+
+  function setCoordinates(e: any) {
+    let canMouseX: number = 0
+    let canMouseY: number = 0
+    let canMouseX1: number
+    let canMouseY1: number
+    let canMouseX2: number
+    let canMouseY2: number
+    if (e?.nativeEvent?.clientX && e?.nativeEvent?.clientY) {
+      //canMouseX = parseInt(e.clientX - cfg.offsetX)
+      canMouseX = e.nativeEvent.clientX - offsetLeft
+      canMouseY = e.nativeEvent.clientY - offsetTop
+    } else if (e?.nativeEvent?.targetTouches) {
+      canMouseX1 = e.nativeEvent.targetTouches.item(0)?.clientX - offsetLeft
+      canMouseY1 = e.nativeEvent.targetTouches.item(0)?.clientY - offsetTop
+      if (e?.nativeEvent?.touches?.length > 1) {
+        //To get the center of the 'pinch' we to 'average' the two points
+        canMouseX2 = e.nativeEvent.targetTouches.item(1)?.clientX - offsetLeft
+        canMouseY2 = e.nativeEvent.targetTouches.item(1)?.clientY - offsetTop
+        canMouseX = Math.round((canMouseX1 + canMouseX2) / 2)
+        canMouseX = Math.round((canMouseY1 + canMouseY2) / 2)
+      } else {
+        canMouseX = canMouseX1
+        canMouseY = canMouseY1
+      }
+    }
+    return {
+      canMouseX,
+      canMouseY,
+    }
+  }
+
+  /* MOUSE TOUCH HANDLERS
+  variables canMouseX/Y refer to the CURRENT coordinates of the mouse or touch for current detected event on the 'canvas'.
+  canMouseX/Y coordinates are compared to the last stored coordinates (oldXCoord,oldYCoord) for purposed of
+  determining the direction of a drag. In the case of pinch, two sets of current coordinates measure a distance between
+  those two points and that to a previously stored distance to determine a pinch "in" or "out"
+  */
+
+  //When Mouse button up (touch released) reset to orginal setting
+  const handleMouseUp = (e: any) => {
+    setIsScaling(false)
+    setIsDragging(false)
+    setPanRate(1)
+    setTouchDist(0)
+  }
+
+  const handleMouseLeave = (e: any) => {
+    //for now, leaving the image does the same as release mouse or touch
+    handleMouseUp(e)
+  }
+
+  const handleMouseDown = (e: any) => {
+    const { canMouseX, canMouseY } = setCoordinates(e)
+    canMouseX ? setOldXCoord(canMouseX) : setOldXCoord(0)
+    canMouseY ? setOldYCoord(canMouseY) : setOldYCoord(0)
+    e.preventDefault()
```

Comment:
So the preventDefault should always come first in most cases, especially since you're performing updates prior to this line of code and that queues a re-render phase.

Reply:
Let's test I was having issues with preventDefault